server/routes/carts.js

```diff
@@ -0,0 +1,113 @@
+const Router = require('express')
+const Cart = require('../models/Cart')
+const router = new Router()
+
+router.post('/api/carts', createCart)
+router.get('/api/carts', getCarts)
+router.get('/api/carts/:id', getCart)
+router.post('/api/carts/:id/products', addProduct)
+router.put('/api/carts/:id/products', changeQuantity)
+router.delete('/api/carts/:id/products', deleteProduct)
+
+async function createCart(req, res) {
+  const cart = req.body
+
+  const newCart = new Cart(cart)
+  try {
+    await newCart.save()
+
+    res.cookie('cart_id', newCart._id.toString())
+    res.status(201).json(newCart)
+  } catch (error) {
+    res.status(409).json({ message: error.message })
+  }
+}
+
+async function getCarts(req, res) {
+  try {
+    const carts = await Cart.find()
+
+    res.status(200).json(carts)
+  } catch (error) {
+    res.status(404).json({ message: error.message })
+  }
+}
+
+async function getCart(req, res) {
+  const { id: _id } = req.params
+
+  try {
+    const cart = await Cart.findById(_id)
+
+    res.json(cart)
+  } catch (error) {
+    res.status(404).json({ message: error.message })
+  }
+}
+
+async function addProduct(req, res) {
+  const product = req.body
+  const { id: _id } = req.params
+
+  try {
+    const cart = await Cart.findById(_id)
+    cart.products = [...cart.products, product]
```

Comment:
@AStashevskaya Also you shouldn't allow use to send custom product fields, if he is an evil hacker, he can add 10000 pizzas with price = 0, or he can buy cat in pizza store, you should just handle `productId` param, and search product in your database:

<img width="1424" alt="image" src="https://user-images.githubusercontent.com/12561302/116731152-a2e01280-a9e9-11eb-9995-57241a5ecc2b.png">